packages/shared-metrics/src/gc.js

```diff
@@ -37,37 +47,42 @@ const nativeModuleLoader = require('./util/nativeModuleRetry')({
 
 let senseIntervalHandle;
 
-exports.activate = function activate() {
-  nativeModuleLoader.on('loaded', gcStats => {
```

Comment:
This was the main issue: We previously only added the listeners to the nativeModuleLoader once `activate` was called (which happens when a connection to the back end is established). This is of course far too late, especially if gcstats.js is already available in node_modules. 

Thus, we only registered the listener long after the `loaded` event would have been emitted, causing us to miss that event entirely and never executing the handler for `loaded`.